### DISCUSSION TITLE
AptosApiError when using aptos.transaction.submit.multiAgent

### DISCUSSION THREAD

**Discussion by MartinYeung5** [aptos-labs/aptos-developer-discussions#372] [Questions] [answered]
Hi team, when using aptos.transaction.submit.multiAgent, it showed the error:

POST https://api.devnet.aptoslabs.com/v1/transactions 400 (Bad Request)
AptosApiError: Request to [Fullnode]: POST https://api.devnet.aptoslabs.com/v1/transactions failed with: {"message":"Invalid transaction: Type: Validation Code: SENDING_ACCOUNT_DOES_NOT_EXIST","error_code":"vm_error","vm_error_code":7}

Tutorial:
https://aptos.guide/en/build/sdks/ts-sdk/building-transactions/multi-agent-transactions

The code:
      import { InputTransactionData, useWallet } from "@aptos-labs/wallet-adapter-react";
      const { account, signAndSubmitTransaction } = useWallet();

      const config = new AptosConfig({ network: Network.DEVNET });
      const aptos = new Aptos(config);

      let bob = Account.generate();
      console.log("bob = " + bob.accountAddress);
      const transfer_amount = 1;
      const transaction = await aptos.transaction.build.multiAgent({
        sender: owner1Address,
        secondarySignerAddresses: [bob.accountAddress],
        data: {
          // REPLACE WITH YOUR MULTI-AGENT FUNCTION HERE
          function: "0x1::aptos_account::transfer_coins",
          functionArguments: ["0x99891379eaa29093a9c517790c75c6859f9dd7780b01e608bc9585bfd374ecec", transfer_amount],
          typeArguments: ["0x1::aptos_coin::AptosCoin"],
        },
      });

      const privateKey = new Ed25519PrivateKey(process.env.REACT_APP_PRIVATE_KEY!);
      const ownerAccount = Account.fromPrivateKey({privateKey});
      console.log(ownerAccount);
      const aliceSenderAuthenticator = aptos.transaction.sign({
        signer: ownerAccount,
        transaction,
      });
      const bobSenderAuthenticator = aptos.transaction.sign({
        signer: bob,
        transaction,
      });

      // 4. Submit
      console.log("\n=== 4. Submitting transaction ===\n");
      const committedTransaction = await aptos.transaction.submit.multiAgent({
        transaction,
        senderAuthenticator: aliceSenderAuthenticator,
        additionalSignersAuthenticators: [bobSenderAuthenticator],
      });

How can i do in this case?

**Comment by gregnazario** [comment, marked as answer]
`SENDING_ACCOUNT_DOES_NOT_EXIST` means that the account that is the first signer account does not exist on chain.

You'll have to create the accounts that sign the function first.

```ts
await aptos.fundAccount({accountAddress: ownerAccount});
await aptos.fundAccount({accountAddress: bob});
```

**Reply by MartinYeung5** [reply]
Actually, i want to let user to sign, how can i do ?

according to the tutorial, it can develop the multisig function, 
and i want to let owner to create the multisig and let specific user to sign it.
therefore, it should let owner login wallet and enter the specific users' wallet, then submit the transaction.
user can read the transaction and sign the multisig function with that transaction.

**Comment by MartinYeung5** [comment]
anyone know how to solve it ? thanks

**Reply by Alexzander-Stone** [reply]
Assuming you followed Greg's advice for creating the accounts on-chain before attempting to craft the TX, you should be able to continue through the tutorial to achieve your goal.

Can you explain where in the tutorial you're having difficulties?

**Reply by MartinYeung5** [reply]
the issue is solved, thanks
